src/jit/lsra.cpp

```diff
@@ -2896,6 +2896,18 @@ regMaskTP LinearScan::getKillSetForNode(GenTree* tree)
                 else
                 {
                     killMask = RBM_INT_CALLEE_TRASH;
+#ifdef _TARGET_ARM_
+                    // On ARM, There is the case that the argument of a virtual stub call is transferred by R4 register.
+                    // And then if the virtual stub is called,
+                    // R4 register would be the busy status to avoid losing the argument data.
+                    // It means R4 register must not be assigned to anywhere until the next kill.
+                    // But the R4 register has to be killed in case of after virtual stub call to escape the busy state.
+                    if (tree->AsCall()->IsVirtualStub())
+                    {
+                        regMaskTP virtualStubParamMask = compiler->virtualStubParamInfo->GetRegMask();
+                        killMask |= virtualStubParamMask;
+                    }
+#endif // _TARGET_ARM_
```

Comment:
This whole clause would be true when `compiler->compFloatingPointUsed` is true, also, so needs to move below the if/else.